### PR TITLE
Make npm deps less scary.  Docs for optional mods

### PR DESCRIPTION
**Pull request by olizilla**
- Adds a brief explanation on how to add optional modules to help people new to the project (me).
- Avoids issues like basscss/bassplate#15 by whittling the npm deps down to core basscss. Accidentally depedending on non-core modules will be easier to spot as poscss will fail.
- Whittling down the npm deps has the added benefit of being way less intimidating to people new to the project, even tho modularity is _a good thing_.

### CODE REVIEW

**Comment by olizilla** [comment]
Also of note, this PR currently pulls out:

``` css
@import 'basscss-utility-typography';
@import 'basscss-utility-layout';
```

from src/base.css as a quick resolution for basscss/bassplate#15, but from the basscss docs it looks like those two are intended to be core modules, and so added as deps to `basscss`. As this is my first play with `basscss` I figured I'd tackle it in a smaller PR rather than meddling with the core.

Anyhoo, just a suggestion, happy to modify the PR if tweaks are needed.

File: package.json [hunk 1]
```diff
@@ -5,49 +5,10 @@
...
-    "basscss-responsive-states": "latest",
-    "basscss-responsive-white-space": "latest",
-    "basscss-table-object": "latest",
-    "basscss-type-scale": "latest",
-    "basscss-ui-utility-groups": "latest",
-    "basscss-utility-headings": "latest",
-    "basscss-utility-layout": "latest",
-    "basscss-utility-typography": "latest",
-    "flex-object": "latest"
+    "basscss": "^7.1.1"
```
Comment (jxnblk) [inline package.json:8]: Correct me if I'm wrong, but this only works with npm 3 and up, right?

Comment (olizilla) [inline package.json:8]: Yes, that is definitely a thing.

**Comment by olizilla** [comment]
Maybe this change is a little premature as node 5 only just started shipping with npm3, but, I'd guess folks who are basscss-ing are also bleeding edge all the things. npm would very much like it if everyone updated any way. 

Not a solid argument tho, so no worries if you think it's too soon. I'm now wondering how this'd play with the monorepo plans...

**Comment by jxnblk** [comment]
Sorry I let this sit for a while. I was trying to get basscss v8 out the door to have a cleaner place to start from.

I still like the doc updates you have here, and I'm totally fine with expecting npm > 3.0.0.

Would you mind seeing if your can realign with master? If it's a PITA, don't worry about it.

**Comment by olizilla** [comment]
@jxnblk no problem! Good work on v8
I'll update this PR.

**Comment by olizilla** [comment]
Closing in favour of #20

**Pull request closed without merging**